sensenova_si/internvl.py

```diff
@@ -0,0 +1,68 @@
+from .model import Model
+from .utils import split_model, load_image
+from transformers import AutoModel, AutoTokenizer
+import torch
+
+class SenseNovaSIInternVLModel(Model):
+    def __init__(self, model_path):
+        self.device_map = split_model(model_path)
+        self.model = AutoModel.from_pretrained(
+            model_path,
+            dtype=torch.bfloat16,
+            # use_flash_attn=True,
+            attn_implementation="flash_attention_2",
+            load_in_8bit=False,
+            low_cpu_mem_usage=True,
+            trust_remote_code=True,
+            device_map=self.device_map,
+        ).eval()
+
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            model_path, trust_remote_code=True, use_fast=False
+        )
+
+        self.default_generation_config = {
+            "do_sample": False,
+            "max_new_tokens": 8192,
+            "top_p": 1.0,
+            "temperature": 0.0,
+            "repetition_penalty": 1,
+            "num_beams": 1,
+        }
```

Comment:
Looks great! How about using `config.yaml` or a global variable for `default_generation_config` to make it easier for users to find and modify?